src/internal/internal.h

```diff
@@ -82,6 +82,7 @@ struct us_socket_t {
     struct us_socket_context_t *context;
     struct us_socket_t *prev, *next;
     unsigned short timeout;
+    unsigned short low_prio_state; /* 0 = not in low-prio queue, 1 = is in low-prio queue, 2 = was in low-prio queue in this iteration */
```

Comment:
If we don't use any other timeout than 0 while in low prio state, can this be moved into timeout variable?

Reply:
These 2 bits can live in timeout all time. 2^14 = 4.5 hours of timeout, that's enough anyways.

Reply:
If I understood your code correctly, the "timeout" holds the "rest" of the current timeout, and is reduced on every `us_internal_timer_sweep()` run in https://github.com/uNetworking/uSockets/blob/master/src/loop.c#L89

As I didn't want to completely reset the timeout, if the socket is in the low-prio-queue, I didn't take this possibility into account. Currently it is just "not reduced" in us_internal_timer_sweep() while it is in the low-priority-queue.

Reply:
> These 2 bits can live in timeout all time.

Haha ... okay, that's nice ;-) Yes we can make it like this:

```
unsigned short timeout : 14;
unsigned short low_prio_state : 2;
```